### PR TITLE
[ADD] Add base_phone module

### PR DESCRIPTION
**Pull request by ehdem**
[ADD] Add base_phone module;it validates phone numbers using the _phonenumbers_ Python library.

This module comes from https://launchpad.net/openerp-asterisk-connector/7.0.

It was done by Alexis de Lattre from Akretion alexis.delattre@akretion.com

### CODE REVIEW

File: partner_firstname/partner.py [hunk 1]
```diff
@@ -145,5 +145,5 @@ def create(self, cursor, uid, vals, context=None):
         ),
 
         'firstname': fields.char("Firstname"),
-        'lastname': fields.char("Lastname", required=True),
+        'lastname': fields.char("Lastname"),
```
Comment (bwrsandman) [inline partner_firstname/partner.py]: I don't think you meant to commit this

File: base_phone/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    'name': 'Base Phone',
+    'version': '0.1',
+    'category': 'Phone',
+    'license': 'AGPL-3',
+    'summary': 'Validate phone numbers',
+    'description': """
+Base Phone
+==========
+
+This module validate phone numbers using the *phonenumbers* Python library,
```
Comment (pedrobaeza) [inline base_phone/__openerp__.py]: s/validate/validates

File: base_phone/base_phone.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+from openerp.osv import orm, fields
+from openerp.tools.translate import _
+import logging
+# Lib for phone number reformating -> pip install phonenumbers
+import phonenumbers
+
+_logger = logging.getLogger(__name__)
+
+
+class phone_common(orm.AbstractModel):
```
Comment (pedrobaeza) [inline base_phone/base_phone.py:31]: Use CamelCase: `PhoneCommon`

File: base_phone/base_phone.py [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+        }
+        return res
+
+    def click2dial(self, cr, uid, erp_number, context=None):
+        '''This function is designed to be overridden in IPBX-specific
+        modules, such as asterisk_click2dial'''
+        return {'dialed_number': erp_number}
+
+
+class res_partner(orm.Model):
```
Comment (pedrobaeza) [inline base_phone/base_phone.py:208]: CamelCase

**Comment by pedrobaeza** [comment]
One flake8 error:

```
./base_phone/wizard/number_not_found.py:80:9: F841 local variable 'number_type' is assigned to but never use
```

**Comment by pedrobaeza** [comment]
Maybe _base_phone_format_ or _partner_phone_format_ can be a more focused name.

File: partner_firstname/partner.py (hunk 1 above)
Comment (ehdem) [inline partner_firstname/partner.py]: yes, you're right

File: base_phone/__openerp__.py (hunk 2 above)
Comment (ehdem) [inline base_phone/__openerp__.py]: yep

File: base_phone/base_phone.py (hunk 3 above)
Comment (ehdem) [inline base_phone/base_phone.py:31]: Why want you use CamelCase  for class name ?

File: base_phone/base_phone.py (hunk 4 above)
Comment (ehdem) [inline base_phone/base_phone.py:208]: Why want you use CamelCase  for class name ?

File: base_phone/base_phone.py (hunk 3 above)
Comment (pedrobaeza) [inline base_phone/base_phone.py:31]: Because they are current conventions (PEP8) for new modules (and desirable for old ones)

**Comment by ehdem** [comment]
New PR: https://github.com/OCA/partner-contact/pull/38/commits

**Pull request closed without merging**